### ISSUE TITLE
Typescript errors when trying to build cloud function

### ISSUE THREAD

**Issue by lsps9150414** [MichaelSolati/geofirestore-js#148]
The errors:
```
➜  functions git:(master) ✗ yarn build
yarn run v1.19.1
$ tsc
node_modules/@types/node/assert.d.ts:47:14 - error TS2300: Duplicate identifier 'internal'.

47     export = internal;
                ~~~~~~~~

  node_modules/geofirestore/node_modules/@types/node/assert.d.ts:47:14
    47     export = internal;
                    ~~~~~~~~
    'internal' was also declared here.

node_modules/@types/node/async_hooks.d.ts:92:11 - error TS2300: Duplicate identifier 'AsyncResource'.

92     class AsyncResource {
             ~~~~~~~~~~~~~

  node_modules/geofirestore/node_modules/@types/node/async_hooks.d.ts:92:11
    92     class AsyncResource {
                 ~~~~~~~~~~~~~
    'AsyncResource' was also declared here.

node_modules/@types/node/buffer.d.ts:2:18 - error TS2451: Cannot redeclare block-scoped variable 'INSPECT_MAX_BYTES'.

2     export const INSPECT_MAX_BYTES: number;
                   ~~~~~~~~~~~~~~~~~

  node_modules/geofirestore/node_modules/@types/node/buffer.d.ts:2:18
    2     export const INSPECT_MAX_BYTES: number;
                       ~~~~~~~~~~~~~~~~~
    'INSPECT_MAX_BYTES' was also declared here.

node_modules/@types/node/buffer.d.ts:3:18 - error TS2451: Cannot redeclare block-scoped variable 'kMaxLength'.

3     export const kMaxLength: number;
                   ~~~~~~~~~~

  node_modules/geofirestore/node_modules/@types/node/buffer.d.ts:3:18
    3     export const kMaxLength: number;
                       ~~~~~~~~~~

... and more
```

My package.json:
```json
{
  "name": "functions",
  "scripts": {
    "lint": "tslint --project tsconfig.json",
    "build": "tsc",
    "serve": "npm run build && firebase serve --only functions",
    "shell": "npm run build && firebase functions:shell",
    "start": "npm run shell",
    "deploy": "firebase deploy --only functions",
    "logs": "firebase functions:log"
  },
  "engines": {
    "node": "8"
  },
  "main": "lib/index.js",
  "dependencies": {
    "change-case": "^3.1.0",
    "firebase-admin": "^8.6.0",
    "firebase-functions": "^3.3.0",
    "geofirestore": "^3.3.1",
    "glob": "^7.1.6",
    "lodash": "^4.17.15"
  },
  "devDependencies": {
    "@types/glob": "^7.1.1",
    "firebase-functions-test": "^0.1.6",
    "tslint": "^5.12.0",
    "typescript": "^3.2.2"
  },
  "private": true
}
```

**Comment by MichaelSolati** [comment]
Looks like a conflict in dependencies, so I'm working on a biggish change in the dev branch, and I was curious if you install geofirestore as so... `yarn add ssh://github.com/geofirestore/geofirestore-js#dev` and then try to build, what happens?

**Comment by lsps9150414** [comment]
@MichaelSolati Thanks for the fast reply. Installing from the link you provide and `yarn build` returns the same error.

**Comment by MichaelSolati** [comment]
Version 3.4.0 is [running through travis now](https://travis-ci.org/geofirestore/geofirestore-js/builds/637738977) and should hopefully resolve this issue. Feel free to give it a shot!

**Comment by lsps9150414** [comment]
@MichaelSolati Awesome! I'll report back when I have time to test it again 👍

**Comment by lsps9150414** [comment]
@MichaelSolati I have recently tested with these dependencies:
```
"dependencies": {
    "change-case": "^3.1.0",
    "child-process-promise": "^2.2.1",
    "copyfiles": "^2.2.0",
    "firebase-admin": "^8.6.0",
    "firebase-functions": "^3.3.0",
    "fs": "^0.0.1-security",
    "geofirestore": "^3.4.1",
    "glob": "^7.1.6",
    "lodash": "^4.17.15",
    "mkdirp": "^0.5",
    "mkdirp-promise": "^4.0.1",
    "os": "^0.1.1",
    "path": "^0.12.7",
    "uuid": "^3.4.0"
  },
```

and I got these errors when trying to build:
```
node_modules/@types/node/globals.d.ts:2:1 - error TS6200: Definitions of the following identifiers conflict with those in another file: BufferEncoding, Buffer, Platform, Signals, MultipleResolveType, BeforeExitListener, DisconnectListener, ExitListener, RejectionHandledListener, UncaughtExceptionListener, UnhandledRejectionListener, WarningListener, MessageListener, SignalsListener, NewListenerListener, RemoveListenerListener, MultipleResolveListener, TypedArray, ArrayBufferView

2 interface Console {
  ~~~~~~~~~

  node_modules/geofirestore/node_modules/@types/node/globals.d.ts:2:1
    2 interface Console {
      ~~~~~~~~~
    Conflicts are in this file.

node_modules/@types/node/assert.d.ts:47:14 - error TS2300: Duplicate identifier 'internal'.

47     export = internal;
                ~~~~~~~~

  node_modules/geofirestore/node_modules/@types/node/assert.d.ts:52:14
    52     export = internal;
                    ~~~~~~~~
    'internal' was also declared here.

node_modules/@types/node/async_hooks.d.ts:92:11 - error TS2300: Duplicate identifier 'AsyncResource'.

92     class AsyncResource {
             ~~~~~~~~~~~~~

  node_modules/geofirestore/node_modules/@types/node/async_hooks.d.ts:92:11
    92     class AsyncResource {
                 ~~~~~~~~~~~~~
    'AsyncResource' was also declared here.

node_modules/@types/node/buffer.d.ts:2:18 - error TS2451: Cannot redeclare block-scoped variable 'INSPECT_MAX_BYTES'.

2     export const INSPECT_MAX_BYTES: number;
                   ~~~~~~~~~~~~~~~~~

  node_modules/geofirestore/node_modules/@types/node/buffer.d.ts:2:18
    2     export const INSPECT_MAX_BYTES: number;
                       ~~~~~~~~~~~~~~~~~
    'INSPECT_MAX_BYTES' was also declared here.

node_modules/@types/node/buffer.d.ts:3:18 - error TS2451: Cannot redeclare block-scoped variable 'kMaxLength'.

3     export const kMaxLength: number;

...
```

**Comment by MichaelSolati** [comment]
Can you post a sample repo with this not working?

**Comment by ZaifSenpai** [comment]
> ```
> cd functions
> npm run-script build
> ```

Even if by index.ts only contain one line:
> ```
> import { GeoFirestore } from 'geofirestore';
> ```

also, in my package.json is:
> ```
> "dependencies": {
>     "firebase-admin": "^8.10.0",
>     "firebase-functions": "^3.6.1",
>     "geofirestore": "^3.4.3"
>   }
> ```

**Comment by MichaelSolati** [comment]
@ZaifSenpai please share a repo where this issue is reproducable.

**Comment by ZaifSenpai** [comment]
> @ZaifSenpai please share a repo where this issue is reproducable.

[Here](https://github.com/ZaifSenpai/test-geofirestore-error)

Just after initializing the firebase app, I installed package via npm, added the dependency in package.json and added import statement in index.ts. Then it showed me the error in deploy.

Maybe this log will help:

```
0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   '--prefix',
1 verbose cli   'C:\\Users\\escbu\\Downloads\\test\\functions',
1 verbose cli   'run',
1 verbose cli   'build' ]
2 info using npm@6.4.1
3 info using node@v11.4.0
4 verbose run-script [ 'prebuild', 'build', 'postbuild' ]
5 info lifecycle functions@~prebuild: functions@
6 info lifecycle functions@~build: functions@
7 verbose lifecycle functions@~build: unsafe-perm in lifecycle true
8 verbose lifecycle functions@~build: PATH: C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin;C:\Users\escbu\Downloads\test\functions\node_modules\.bin;C:\Perl64\site\bin;C:\Perl64\bin;C:\Python36\Scripts\;C:\Python36\;C:\ProgramData\Oracle\Java\javapath;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Java\jdk1.8.0_65\bin;C:\Java\jre1.8.0_65;C:\Java\jre1.8.0_65\bin;C:\Java\jre1.8.0_65\lib;C:\Program Files\Sublime Text 3;C:\Program Files (x86)\MKVToolNix;C:\Program Files (x86)\QuickTime\QTSystem\;C:\Program Files\ggobi;C:\Program Files\dotnet\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\Client SDK\ODBC\110\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\120\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\120\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\120\Tools\Binn\ManagementStudio\;C:\Program Files\PuTTY\;C:\Program Files (x86)\OpenSSH\bin;C:\ANDROID_SDK\platform-tools;C:\Program Files\C,C++ Compiler\bin;%BOOST_ROOT%;C:\Program Files\Git\cmd;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\vcpkg;C:\Program Files (x86)\PDFtk\bin\;C:\Program Files\nodejs\;C:\Program Files (x86)\EaseUS\Todo Backup\bin;;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\xampp\php;C:\ProgramData\ComposerSetup\bin;C:\Program Files\Microsoft Network Monitor 3\;C:\Program Files\MySQL\MySQL Shell 8.0\bin\;C:\Program Files\Microsoft VS Code\bin;C:\Users\escbu\AppData\Local\GitHubDesktop\bin;C:\Program Files\CyberGhost 7;C:\Program Files\Tesseract-OCR;C:\Users\escbu\AppData\Roaming\Composer\vendor\bin;C:\Program Files\JetBrains\PhpStorm 2019.3.1\bin;C:\Users\escbu\AppData\Roaming\npm;C:\Program Files\JetBrains\WebStorm 2019.2\bin;C:\Program Files\JetBrains\CLion 2019.3.3\bin
9 verbose lifecycle functions@~build: CWD: C:\Users\escbu\Downloads\test\functions
10 silly lifecycle functions@~build: Args: [ '/d /s /c', 'tsc' ]
11 silly lifecycle functions@~build: Returned: code: 2  signal: null
12 info lifecycle functions@~build: Failed to exec build script
13 verbose stack Error: functions@ build: `tsc`
13 verbose stack Exit status 2
13 verbose stack     at EventEmitter.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\index.js:301:16)
13 verbose stack     at EventEmitter.emit (events.js:189:13)
13 verbose stack     at ChildProcess.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:189:13)
13 verbose stack     at maybeClose (internal/child_process.js:978:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:265:5)
14 verbose pkgid functions@
15 verbose cwd C:\Users\escbu\Downloads\test
16 verbose Windows_NT 10.0.18363
17 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "--prefix" "C:\\Users\\escbu\\Downloads\\test\\functions" "run" "build"
18 verbose node v11.4.0
19 verbose npm  v6.4.1
20 error code ELIFECYCLE
21 error errno 2
22 error functions@ build: `tsc`
22 error Exit status 2
23 error Failed at the functions@ build script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 2, true ]

```

and I have attached the full redirected output of "firebase deploy" command
[output.txt](https://github.com/MichaelSolati/geofirestore-js/files/4637896/stdout.txt)

**Comment by MichaelSolati** [comment]
Odd. So I ran a build and only got the following error:

```
src/index.ts:1:1 - error TS6133: 'GeoFirestore' is declared but its value is never read.

1 import { GeoFirestore } from 'geofirestore';
```

I'm curious if this is a Windows issue. I don't have access to a windows machine so it may take a while to address this. Can you just run `tsc` and paste the result?

**Comment by ZaifSenpai** [comment]
> @ZaifSenpai
> 
> Odd. So I ran a build and only got the following error:
> 
> ```
> src/index.ts:1:1 - error TS6133: 'GeoFirestore' is declared but its value is never read.
> 
> 1 import { GeoFirestore } from 'geofirestore';
> ```
> 
> I'm curious if this is a Windows issue. I don't have access to a windows machine so it may take a while to address this. Can you just run `tsc` and paste the result?

This is the output of the command "tsc index.ts": 
[cmd.txt](https://github.com/MichaelSolati/geofirestore-js/files/4639038/cmd.txt)

**Comment by ZaifSenpai** [comment]
@MichaelSolati 
I tried it on the built in ubuntu on windows 10, same problem. Maybe the issue is with versions? This is my first time using Typescript by the way.

On cmd:
`tsc v3.9.2`
`node v11.4.0`
`npm v6.4.1`

On bash:
`tsc v3.9.2`
`node v12.16.3`
`npm v6.14.4`

**Comment by ZaifSenpai** [comment]
@MichaelSolati 
Fixed my problem!

First I updated node to latest version: 
`node v14.2.0`
`npm v6.14.5`
(I also installed the 'additional libraries' which nodejs installer asks you to install)
and re-installed tslint:
`npm install tslint typescript -g `

Removed `node_modules` directory from my Firebase Cloud Functions project directory, and re-installed them by:
```
cd functions
npm install
```

That's all! It created the `node_modules` directory and `firebase deploy` worked!
